proposals/declarative-secrets.md

```diff
@@ -0,0 +1,58 @@
+# Declarative Secrets
+
+## Motiviation
+
+Various projects exist in the wild that require various secrets for them to run properly. Exmaples include:
+
+- https://github.com/lostintangent/codespaces-langchain
+- https://github.com/openai/openai-quickstart-python
+- https://github.com/openai/openai-quickstart-node
+
+Today these projects have to include specific instructions e.g. in their README telling users where to procure what they need to run and then how to set it up as a secret or add it as an environment variable. This currently acts as an impediment to adoption and promotion of codespaces for these projects.
+
+## Goal
+
+Simplify using codespaces for these kinds of projects by including it as a first-class part of the codespace creation flow.
+
+## Proposal
+
+Add an optional `secrets` to the `devContainer.base.schema.json`. This will be used to declare the secrets needed within the devcontainer.
```

Comment:
nit: We refer to them as 2 words "dev containers," so would be great to update here and any others spots in the doc 😄